File: Grokking Algorithms in C++/Chapter 1 .cpp
#include <iostream>
using namespace std;

int binarySearch(int arr[], int N, int item)
{
	int low = 0, high = N - 1;
	while (low <= high)
	{
		int mid = (high + low) / 2;	//access the middle item
		if (arr[mid] == item)
		{
			return mid;
		}
		else if (arr[mid] > item)
			high = mid - 1;
		else
			low = mid + 1;
	}
	return -1;
}

int main()
{
	//Remember: the array must be sorted
	int myArray[10] = { 1,2,3,4,5,6,7,8,9,10 };
	
	cout << binarySearch(myArray, 10,5) << endl;	//prints: 4
	cout << binarySearch(myArray, 10, 89) << endl;	//prints: -1

	return 0;
}

File: Grokking Algorithms in Python/Chapter 1 .py
#BINARY SEARCH

#define a function to find the item using the binary search technique
def binarySearch(list, item):
    low = 0
    high = len(list) - 1
    while low <= high:
        mid = (low + high) // 2    #go to the item in the half            # // ===> means floor division
        guess = list[mid]          #guess is the found number
        if guess == item:
            return mid
        elif guess > item:         #if guess bigger than item, search in the left side
            high = mid - 1
        else:
            low = mid + 1          #if guess smaller then item, search in the right side
    return None


#test the function
#Note: the list must be sorted
myList = [1,2,3,4,5,6,7,8,9,10]
print(binarySearch(myList, 9))              #prints: 8
print(binarySearch(myList, 15))             #prints: None
